cmd/tuf/app/init.go

```diff
@@ -205,6 +206,24 @@ func InitCmd(ctx context.Context, directory, previous string, threshold int, tar
 		if err := repo.AddTargetWithExpiresToPreferredRole(base, custom, getExpiration("targets"), "targets"); err != nil {
 			return fmt.Errorf("error adding targets %w", err)
 		}
+		expectedTargets[base] = true
+	}
+
+	// Remove old targets that were not included in config.
+	targetsToRemove := []string{}
+	allTargets, err := repo.Targets()
+	if err != nil {
+		return err
+	}
+	for path := range allTargets {
+		if !expectedTargets[path] {
+			targetsToRemove = append(targetsToRemove, path)
+		}
+	}
+	if len(targetsToRemove) > 0 {
+		if err := repo.RemoveTargetsWithExpires(targetsToRemove, getExpiration("targets")); err != nil {
```

Comment:
interesting! does `RemoveTargetsWithExpires` with an empty slice of targets end up removing all targets (is this what you were mentioning?)

if so can you add a quick comment like "only remove with non-empty targets. passing an empty slice removes all targets"

Reply:
That's exactly what it does, will add a comment.